src/rpcrawtransaction.cpp

```diff
@@ -24,111 +23,280 @@ extern std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blo
+            return res;
         }
     }
 
     //Deserialize
     MiningCPID bb = DeserializeBoincBlock(block.vtx[0].hashBoinc,block.nVersion);
 
-    res["height"]=pindex->nHeight;
-    res["version"]=block.nVersion;
+    res.push_back(std::make_pair("Height", ToString(pindex->nHeight)));
+    res.push_back(std::make_pair("Block Version", ToString(block.nVersion)));
+    res.push_back(std::make_pair("Difficulty", RoundToString(GetBlockDifficulty(block.nBits),8)));
+    res.push_back(std::make_pair("CPID", bb.cpid));
+    res.push_back(std::make_pair("Interest", RoundToString(bb.InterestSubsidy,8)));
 
-    res["cpid"]=bb.cpid;
-    res["Magnitude"]=bb.Magnitude;
-    res["Interest"]=bb.InterestSubsidy;
-    res["BoincReward"]=bb.ResearchSubsidy;
-    res["Difficulty"]=GetBlockDifficulty(block.nBits);
+    if (bb.Magnitude > 0)
+    {
+        res.push_back(std::make_pair("Magnitude", RoundToString(bb.Magnitude,8)));
+        res.push_back(std::make_pair("Boinc Reward", RoundToString(bb.ResearchSubsidy,8)));
+    }
 
+    res.push_back(std::make_pair("Is Superblock", (bb.superblock.length() >= 20 ? "true" : "false")));
 
     if(fDebug)
     {
-        res["xbbBoincPublicKey"]=bb.BoincPublicKey;
+        if (bb.superblock.length() >= 20)
+            res.push_back(std::make_pair("Neural Contract Binary Size", ToString(bb.superblock.length())));
+
+        res.push_back(std::make_pair("Organization", bb.Organization));
+        res.push_back(std::make_pair("Client Version", bb.clientversion));
+        res.push_back(std::make_pair("Neural Hash", bb.NeuralHash));
+        res.push_back(std::make_pair("CurrentNeuralHash", bb.CurrentNeuralHash));
+        res.push_back(std::make_pair("Boinc Public Key", bb.BoincPublicKey));
+    }
 
-        res["xbbOrganization"]=bb.Organization;
-        res["xbbClientVersion"]=bb.clientversion;
+    return res;
+}
 
-        res["xbbNeuralHash"]=bb.NeuralHash;
-        res["xbbCurrentNeuralHash"]=bb.CurrentNeuralHash;
-        res["xbbNeuralContractSize"]=bb.superblock.length();
-    }
-    else
+std::vector<std::pair<std::string, std::string>> GetTxNormalBoincHashInfo(const CMerkleTx& mtx)
+{
+    assert(!mtx.IsCoinStake() && !mtx.IsCoinBase());
+    std::vector<std::pair<std::string, std::string>> res;
+
+    try
     {
-        if(bb.superblock.length()>=20)
+        const std::string &msg = mtx.hashBoinc;
+
+        res.push_back(std::make_pair("Network Time", TimestampToHRDate((double)mtx.nTime)));
+
+        if (fDebug)
+            res.push_back(std::make_pair("Message Length", ToString(msg.length())));
+
+        std::string sMessageType = ExtractXML(msg, "<MT>", "</MT>");
+        std::string sTxMessage = ExtractXML(msg, "<MESSAGE>", "</MESSAGE>");
+        std::string sRainMessage = ExtractXML(msg, "<NARR>", "</NARR>");
+
+        if (sMessageType.length())
+        {
+            res.push_back(std::make_pair("Message Type", "Contract"));
```

Comment:
I am not comfortable with info like this:
```
Message Type: Contract
Contract Type: Beacon
Contract Commit: Add
```
Can you replace it with something like:
```
Message Type: Add Beacon Contract
```
This saves quite a bit of vertical space and makes the interface look less like field enumeration.

Reply:
i'll work something up after dinner.

Reply:
how about Message Type: Beacon Contract and Action: Add @denravonska ?

Reply:
-- complete